plugins/obs-qsv11/obs-qsv11.c

```diff
@@ -962,6 +1162,71 @@ static void parse_packet(struct obs_qsv *obsqsv, struct encoder_packet *packet,
 	g_bFirst = false;
 }
 
+static void parse_packet_av1(struct obs_qsv *obsqsv,
+			     struct encoder_packet *packet, mfxBitstream *pBS,
+			     const struct video_output_info *voi,
+			     bool *received_packet)
+{
+	if (pBS == NULL || pBS->DataLength == 0) {
+		*received_packet = false;
+		return;
+	}
+
+	da_resize(obsqsv->packet_data, 0);
+	da_push_back_array(obsqsv->packet_data, &pBS->Data[pBS->DataOffset],
+			   pBS->DataLength);
+
+	packet->data = obsqsv->packet_data.array;
+	packet->size = obsqsv->packet_data.num;
+	packet->type = OBS_ENCODER_VIDEO;
+	packet->pts = ts_mfx_to_obs((mfxI64)pBS->TimeStamp, voi);
+	packet->keyframe = (pBS->FrameType & MFX_FRAMETYPE_IDR);
+
+	uint16_t frameType = pBS->FrameType;
+	uint8_t priority;
+
+	if (frameType & MFX_FRAMETYPE_I)
+		priority = OBS_NAL_PRIORITY_HIGHEST;
+	else if ((frameType & MFX_FRAMETYPE_P) ||
+		 (frameType & MFX_FRAMETYPE_REF))
+		priority = OBS_NAL_PRIORITY_HIGH;
+	else
+		priority = OBS_NAL_PRIORITY_DISPOSABLE;
+
+	packet->priority = priority;
+
+	bool pFrame = pBS->FrameType & MFX_FRAMETYPE_P;
+
+	// In case MSDK doesn't support automatic DecodeTimeStamp, do manual
+	// calculation
+	if (g_pts2dtsShift >= 0) {
+		if (g_bFirst) {
+			packet->dts = packet->pts - 3 * obsqsv->params.nFpsDen;
+		} else if (pFrame) {
+			packet->dts = packet->pts - 10 * obsqsv->params.nFpsDen;
+			g_prevDts = packet->dts;
+		} else {
+			packet->dts = g_prevDts + obsqsv->params.nFpsDen;
+			g_prevDts = packet->dts;
+		}
+	} else {
```

Comment:
Not yet. We’re actually refining automatic DTS function in terms of accuracy. Please use existing manual solution until we complete it.

Reply:
What I'm saying is that `g_pts2dtsShift` is defined here:
* https://github.com/obsproject/obs-studio/blob/28.1.2/plugins/obs-qsv11/obs-qsv11.c#L669-L686

Which suggests that this will never actually use the manual calculation in the AV1 branch, because AV1 is not supported in MSDK 1.6 or lower.

Reply:
Sorry for the misunderstanding! You are correct and we had a copy-paste error on our side. It should be fixed now.